### PR TITLE
Add game controls example

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Adds a new example for moving around the map with game-like controls as described in #2943  
Can you please review, @mollymerp and @lucaswoj ?

### CODE REVIEW

File: docs/_posts/examples/3400-01-26-game-controls.html [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    var deltaDegrees = 25; // rotation of the map in degrees when the left or right arrow is clicked
+
+    // prevent zooming and panning that is not done
+    // with the keyboard arrows or buttons
+    map.boxZoom.disable();
+    map.dragPan.disable();
+    map.doubleClickZoom.disable();
+    map.scrollZoom.disable();
+    map.keyboard.disable();
+    map.touchZoomRotate.disable();
```
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-26-game-controls.html]: This block of code disables all interaction handlers except `dragRotate`. Do you intend to leave `dragRotate` enabled? Can you use the [`interactive: false` constructor option](https://www.mapbox.com/mapbox-gl-js/api/#Map) instead?

File: docs/_posts/examples/3400-01-26-game-controls.html [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    function buttonEnd() {
+        clearInterval(persist);
+    }
+
+    buttons.forEach(function(b) {
+        b[1].addEventListener('mousedown', buttonStart.bind(this, b));
+        b[1].addEventListener('touchstart', buttonStart.bind(this, b));
+        b[1].addEventListener('mouseup', buttonEnd.bind(this, b));
+        b[1].addEventListener('touchend', buttonEnd.bind(this, b));
+    });
```
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-26-game-controls.html]: Do we really need buttons AND keyboard shortcuts? A lot of code could be eliminated & simplified if we dropped the buttons.

File: docs/_posts/examples/3400-01-26-game-controls.html [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            map.easeTo({
+                bearing: pos,
+                easing: easeTo
+            });
+        } else {
+            map.panBy(pos, {
+                easing: easeTo
+            });
+        }
+    }
```
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-26-game-controls.html]: This `move` function does not provide any abstraction over the `Map#easeTo` method. Can we eliminate this function in favor of `Map#easeTo`? Note that `Map#panBy` calls `Map#easeTo` internally.

File: docs/_posts/examples/3400-01-26-game-controls.html [hunk 4]
```diff
@@ -0,0 +1,133 @@
+---
+layout: example
+category: example
+title: Game-like controls
```
Comment (lucaswoj) [inline docs/_posts/examples/3400-01-26-game-controls.html]: The example title should be a verb phrase

 > Navigate the map with game-like controls

**Comment by colleenmcginnis** [comment]
Addressed @lucaswoj 's comments including taking away the buttons (so the user only navigates with the keyboard). Since you need to focus on the map before the arrow keys will work, I added one line of instruction on top of the map that disappears once the user clicks on the map. 
image

**Comment by mollymerp** [comment]
Nice work @colleenmcginnis! 

One note: you may want to get rid of `goDirection` entirely, and just put the `map.easeTo` and `map.panBy` calls inside the event listener function (where you currently call `goDirection()` )to make the example even more straight forward.

**Comment by lucaswoj** [comment]
Just pushed up some minor comment edits 🙇 

This is ready to merge on 🍏!

File: docs/_posts/examples/3400-01-26-game-controls.html [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            });
+        } else if (e.which === 40) { // down
+            map.panBy([0, deltaDistance], {
+                easing: easeFunction
+            });
+        } else if (e.which === 37) { // left
+            map.easeTo({
+                bearing: map.getBearing() - deltaDegrees,
+                easing: easeFunction,
+                animate: true
```
Comment (mollymerp) [inline docs/_posts/examples/3400-01-26-game-controls.html]: also you can remove `animate: true` on your `map.eastTo()` calls -- its `true` by defailt 😸

Comment (colleenmcginnis) [inline docs/_posts/examples/3400-01-26-game-controls.html]: done @mollymerp !

**Comment by mollymerp** [comment]
Awesome! Feel free to merge on 🍏 @colleenmcginnis 👏

**Pull request merged**